contracts/ERC4626/VenusERC4626Factory.sol

```diff
@@ -0,0 +1,104 @@
+
+    /// @notice Emitted when a new ERC4626 vault has been created
+    /// @param vToken The vToken used by the vault
+    /// @param vault The vault that was created
+    event CreateERC4626(VTokenInterface indexed vToken, ERC4626Upgradeable indexed vault);
+
+    /// @notice Emitted when the reward recipient address is updated.
+    /// @param oldRecipient The previous reward recipient address.
+    /// @param newRecipient The new reward recipient address.
+    event RewardRecipientUpdated(address indexed oldRecipient, address indexed newRecipient);
+
+    /// @notice Thrown when the provided comptroller is invalid (not registered in PoolRegistry)
+    error VenusERC4626Factory__InvalidComptroller();
+
+    /// @notice Initializes the contract
+    /// @param accessControlManager Address of the ACM contract
+    /// @param poolRegistryAddress Address of the Pool Registry contract
+    /// @param rewardRecipientAddress Reward recipient address
+    /// @param venusERC4626Implementation Address of the VenusERC4626 implementation contract
+    /// @param loopsLimitNumber The loops limit for the MaxLoopsLimit helper
+    function initialize(
+        address accessControlManager,
+        address poolRegistryAddress,
+        address rewardRecipientAddress,
+        address venusERC4626Implementation,
+        uint256 loopsLimitNumber
+    ) external initializer {
+        // checks are missing
+        __Ownable2Step_init();
+        __AccessControlled_init_unchained(accessControlManager);
+
+        poolRegistry = PoolRegistryInterface(poolRegistryAddress);
+        rewardRecipient = rewardRecipientAddress;
+        loopsLimit = loopsLimitNumber;
+
+        // Deploy the upgradeable beacon with the initial implementation
+        beacon = new UpgradeableBeacon(venusERC4626Implementation);
+    }
+
+    /// @notice Sets a new reward recipient address
+    /// @param newRecipient The address of the new reward recipient
+    /// @custom:access Requires access control
+    /// @custom:error ZeroAddressNotAllowed is thrown when the new recipient address is zero
+    /// @custom:event RewardRecipientUpdated is emitted when the reward recipient address is updated
+    function setRewardRecipient(address newRecipient) external {
+        _checkAccessAllowed("setRewardRecipient(address)");
+        ensureNonzeroAddress(newRecipient);
+
+        emit RewardRecipientUpdated(rewardRecipient, newRecipient);
+        rewardRecipient = newRecipient;
+    }
+
+    /// @notice Creates a VenusERC4626 vault for a given asset and comptroller
+    /// @param vToken The vToken address to create the vault
+    /// @return vault The deployed VenusERC4626 vault
+    function createERC4626(address vToken) external returns (ERC4626Upgradeable vault) {
+        ensureNonzeroAddress(vToken);
+
+        VTokenInterface vToken_ = VTokenInterface(vToken);
+
+        if (poolRegistry.getPoolByComptroller(address(vToken_.comptroller())).comptroller == address(0)) {
```

Comment:
I would check this:

```
vToken == poolRegistry.getVTokenForAsset(vToken_.comptroller(), vToken_.underlying());
```

To be sure the provided vToken is included in a valid pool registered in our registry. With the current check, any contract returning a valid comptroller in the `comptroller()` function would pass the check.

Reply:
Used `getVTokenForAsset` instead of `getPoolByComptroller` to verify.